Clean up E0752 explanation (#75482)
rust-lang/rust: src/librustc_error_codes/error_codes/E0752.md

The entry point of the program was marked as `async`.

Erroneous code example:

```compile_fail,E0752
async fn main() -> Result<i32, ()> { // error!
    Ok(1)
}
```

> `fn main()` or the specified start function is not allowed to be `async`. You
> might be seeing this error because your async runtime library is not set up
> correctly. To fix it, don't declare the entry point as `async`:

Reviewer: ```suggestion
`fn main()` or the specified start function is not allowed to be `async`. Not
having a correct async runtime library setup may cause this error. To fix it,
declare the entry point without `async`:
```

Change to active wording, remove the use of "you".

Do we need "the specified start function"? Is `start` redirected to this error too?

```
fn main() -> Result<i32, ()> { // ok!
    Ok(1)
}

> ```

Reviewer: We should put up 3rd party suggestions in case a beginner stumbled up upon this, for example `#[tokio::runtime]`.

Author: We mention it, but I'm not sure it's a good idea to bring a specific library into a "general" example.

Reviewer: Where did we mentioned it? I don't quite like bringing a specific library here too but I believe beginners might not understand what is an async runtime, I first heard of the concept async runtime when I use rust so I do think we should point them somewhere. I don't have a good idea how but that is my suggestion.

Author: I think at this point, the error itself is clear enough. Imagine having an error with `smol` and seeing this explanation talk about `tokio`, it would be more confusing than helpful for newcomers I think.
